tests/route/test_route_map.py

```diff
@@ -16,10 +17,20 @@
 RouteMapTestCase = Tuple[str, str, Set[str]]
 
 
-def is_path_in_route_map(route_map: Dict[str, Any], path: str, path_params: Set[str]) -> bool:
+def is_path_in_route_map(route_map: RouteMapDict, path: str, path_params: Set[str]) -> bool:
     if not path_params:
         return path in route_map
-    components = ["/", *[component for component in param_pat.sub("*", path).split("/") if component]]
+    components = cast(
+        "List[Union[str, _PathParam]]",
+        [
+            "/",
+            *[
+                _path_param if param_pat.fullmatch(component) else component
```

Comment:
Can you rename `param_pat` to `param_pattern`? This is somewhat confusing (I know its not new, but while at it..)